src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimResponseFormGeneratorForSpec.java

```diff
@@ -32,24 +35,44 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import static uk.gov.hmcts.reform.civil.service.robotics.utils.RoboticsDataUtil.CIVIL_COURT_TYPE_ID;
+
 @Service
 @RequiredArgsConstructor
-public class SealedClaimResponseFormGeneratorForSpec implements TemplateDataGenerator<SealedClaimResponseFormForSpec> {
+public class SealedClaimResponseFormGeneratorForSpec implements TemplateDataGeneratorWithAuth<SealedClaimResponseFormForSpec> {
 
     private final RepresentativeService representativeService;
     private final DocumentGeneratorService documentGeneratorService;
     private final DocumentManagementService documentManagementService;
+    private final LocationRefDataService locationRefDataService;
 
     @Override
-    public SealedClaimResponseFormForSpec getTemplateData(CaseData caseData) {
+    public SealedClaimResponseFormForSpec getTemplateData(CaseData caseData, String authorisation) {
+        String requestedCourt = null;
+        StatementOfTruth statementOfTruth = null;
+        if (caseData.getRespondent1DQ().getRespondent1DQRequestedCourt() != null) {
+            requestedCourt = caseData.getRespondent1DQ().getRespondent1DQRequestedCourt().getCaseLocation().getBaseLocation();
+            statementOfTruth = caseData.getRespondent1DQ().getRespondent1DQStatementOfTruth();
+        } else if (caseData.getRespondent2DQ().getRespondent2DQRequestedCourt() != null) {
+            requestedCourt = caseData.getRespondent2DQ().getRespondent2DQRequestedCourt().getCaseLocation().getBaseLocation();
+            statementOfTruth = caseData.getRespondent2DQ().getRespondent2DQStatementOfTruth();
+        }
+        List<LocationRefData> courtLocations = (locationRefDataService
+            .getCourtLocationsByEpimmsId(
+                authorisation,
+                requestedCourt));
         SealedClaimResponseFormForSpec.SealedClaimResponseFormForSpecBuilder builder
             = SealedClaimResponseFormForSpec.builder()
             .referenceNumber(caseData.getLegacyCaseReference())
             .caseName(DocmosisTemplateDataUtils.toCaseName.apply(caseData))
             .whyDisputeTheClaim(caseData.getDetailsOfWhyDoesYouDisputeTheClaim())
-            .statementOfTruth(caseData.getRespondent1DQ().getRespondent1DQStatementOfTruth());
+            .hearingCourtLocation(courtLocations.isEmpty() ? null : courtLocations.stream()
+                .filter(id -> id.getCourtTypeId().equals(CIVIL_COURT_TYPE_ID))
+                .collect(Collectors.toList()).get(0).getCourtName())
```

Comment:
same here please

Reply:
I make changes and make some manual test.